docs/content/sips/018-application-metadata.md

```diff
@@ -0,0 +1,73 @@
+| Component id                                  | Logging (?)      |       |
+| Files mounted to the component                | Not given        | Already available by traversing the component's file system |
+| Allowed outbound hosts for the component      | Not given        | One user mentioned their app needed to scan a set of sites but I don't have details |
+| Allowed KV/SQLite stores for the component    | Explorer UI      |       |
+| Component HTTP route                          | Not given        | Already available via the `spin-raw-component-route` header |
+| The component's build directory               | Guest files mapped at `workdir` | Not clear why user mapped to the subdirectory |
+
+## Should we?
+
+We've been a bit wary of doing some of this stuff because the Wasm model is that components should define the set of capabilities they consume, and be granted only those capabilities, so there should be no need for them to be going around sniffing to find out what goodies they have access to. This concern applies specifically to the "allowed" lists (outbound hosts and KV/SQLite). Possibly this would be addressed by requiring components to declare access.
+
+Also, I'd argue we should _not_ provide access to `workdir`. This is, I believe, not currently available at runtime; and in any case it should have nothing to do with what happens at runtime - build is way distant in the rear view mirror by then. This is better handled by mapping files up a level or by passing a distinct "file root" variable.
+
+| Field                                         | Recommendation   |
+|-----------------------------------------------|------------------|
+| Application name and version                  | Unproblematic    |
+| Host runtime name or other identifying string | Unproblematic    |
+| Host runtime version                          | Unproblematic    |
+| Component id                                  | Unproblematic    |
+| Files mounted to the component                | Superfluous      |
+| Allowed outbound hosts for the component      | Caution advised  |
+| Allowed KV/SQLite stores for the component    | Caution advised  |
+| Component HTTP route                          | Superfluous      |
+| The component's build directory               | Should not       |
+
+# User experience
+
+## Access
+
+If we do wish to expose this data, there are various ways we could do so:
+
+1. Provide read access to the manifest file itself. This couples application logic to the manifest format du jour. We should not do this.
+
+2. Create a new API that reads specific fields e.g. `application_meta::component_id()`. (A variant would be a single function accepting a field ID, e.g. `application_meta::get(AppMeta::ComponentId)`.) This is safe and discoverable, but does not line up with any broader conventions, and may have versioning implications if we need to add additional fields over time.
```

Comment:
Nit: I think we should mention a possible middle ground between complete access to the manifest and access to individual fields -  we could bundle a certain subset of manifest data together and provide one new function that provides access to that data.